src/syscheckd/seechanges.c

```diff
@@ -310,7 +310,7 @@ char *seechanges_addfile(const char *filename)
     char old_location[PATH_MAX + 1];
     char tmp_location[PATH_MAX + 1];
     char diff_location[PATH_MAX + 1];
-    char diff_cmd[PATH_MAX + 1];
+    char diff_cmd[PATH_MAX + OS_SIZE_1024];
```

Comment:
Expanding `diff_cmd[4095]` to `diff_cmd[5120]` is useless because `snprintf` is limited to 2048.
I think the correct solution is to leave this line as it was, and replace `2048` with `sizeof(diff_cmd)`.